### PR TITLE
Fix zabbix-agent tlspsk_auto to support become on Linux and ignore on windows

### LINKED ISSUE #533: AutoTLSPSK configuration is broken
##### SUMMARY
https://github.com/ansible-collections/community.zabbix/pull/495 has broken autotlspsk configuration for the role

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
https://github.com/ansible-collections/community.zabbix/blob/main/roles/zabbix_agent/tasks/tlspsk_auto.yml#L48-L58

##### ANSIBLE VERSION

```paste below
ansible [core 2.11.6]
  config file = None
  configured module search path = ['/home/crepe/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/crepe/venv/ansible/lib/python3.9/site-packages/ansible
  ansible collection location = /home/crepe/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/crepe/venv/ansible/bin/ansible
  python version = 3.9.6 (default, Jun 28 2021, 08:57:49) [GCC 10.3.0]
  jinja version = 3.0.2
  libyaml = True
```

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT / Zabbix Version
N/A

##### STEPS TO REPRODUCE
Run the role with the following parameters defined
```
zabbix_agent_tlsaccept: psk
zabbix_agent_tlsconnect: psk
zabbix_agent_tlspsk_auto: True
```

OR run this playbook. Reference his to see where I got this from: https://github.com/ansible-collections/community.zabbix/blob/main/roles/zabbix_agent/tasks/tlspsk_auto.yml#L48-L58

```yaml
---
- name: Test
  hosts: localhost
  connection: local
  vars:
    zabbix_agent_hostname: "{{ inventory_hostname }}"
    zabbix_agent_visible_hostname: "{{ inventory_hostname }}"
    zabbix_agent2: True
    zabbix_agent2_hostname: "{{ inventory_hostname }}"
  tasks:
    - name: AutoPSK | Generate new TLS PSK identity
      set_fact:
        zabbix_agent_tlspskidentity: >
          {{
            zabbix_agent_visible_hostname
            | ((zabbix_agent2 == True) | default(ternary(zabbix_agent2_hostname, zabbix_agent_hostname)))
            + '_'
            + lookup('password', '/dev/null chars=hexdigits length=4')
          }}
    - debug:
        msg: "{{ zabbix_agent_tlspskidentity }}"
```

##### EXPECTED RESULTS
TLS PSK is configured correctly

##### ACTUAL RESULTS
The role fails to run with the following error
```
fatal: [localhost]: FAILED! => {"msg": "template error while templating string: expected token 'name', got '('. String: {{ \n  zabbix_visible_hostname\n  | ((zabbix_agent2 == true ) | default(ternary(zabbix_agent2_hostname, zabbix_agent_hostname)))\n  + '_'\n  + lookup('password', '/dev/null chars=hexdigits length=4')\n}}\n"}
```

I have submitted a merge request which fixes this issue: #534 

Also I would suggest a version be released so that my team can pin to that version

### PR DESCRIPTION
**Pull request by jeremybusk**
##### SUMMARY
- Windows had issues with tlspsk auto because of become command
- So fixed tlspsk_auto by not using "become" for sudo on Windows hosts
- I added common file for tlspsk_auto & reorganized includes to support above
- Common file is common or shared tasks to include in Linux & Windows files that are called in tlspsk_auto file
- Included https://github.com/ansible-collections/community.zabbix/issues/534 which Fixes #533 #534 for the generated condition as I had issues with it as well.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
zabbix_agent (tlspsk_auto)

##### ADDITIONAL INFORMATION
This may or may not be relavant but here are the specs from windows test I was running
- Ansible 2.11
- Python 3.6
- Remote Powershell version 5.1

I was tempted to use not_windows instead of linux for file name prefix but just stayed with linux. Let me know if I should change.

One last note, maybe it will help someone.

In Zabbix 5.4 tlspsk keys become write only through api (as they should be) 
https://www.zabbix.com/documentation/current/manual/introduction/whatsnew540

https://github.com/ansible-collections/community.zabbix/blob/2f1898f36ba2c8751e84c2892aa04fd003e52f6b/plugins/modules/zabbix_host.py#L834 

Zabbix Server, api doesn't return existing tls_psk/identity value in host object to do diff/changes and Return true in line above

You need to set var so unencrypted and then rerun encrypted for force key change  (# is so remove/comment out)
```
zabbix_agent2_tlsaccept: unencrypted
# zabbix_agent2_tlspsk_auto: True
# zabbix_agent2_tlsaccept: psk
```
and then run with vars set like
```
# zabbix_agent2_tlsaccept: unencrypted
zabbix_agent2_tlspsk_auto: True
zabbix_agent2_tlsaccept: psk
```

### CODE REVIEW

**Comment by doyl54** [comment]
Removed "conf" in Windows_conf.yml so we look for conf file in {{ zabbix_win_install_dir }} and not in conf from zip archive. I think this desired, looking for guidance if not.

Nope this conf has to be here or in vars but it has to be somewhere or conf file is not at the right place

**Comment by jeremybusk** [comment]
@doyl54 Thanks for the guidance on conf in path. II removed that and it looks like everything is working as expected now.

File: roles/zabbix_agent/tasks/tlspsk_auto.yml [hunk 1]
```diff
@@ -3,75 +3,7 @@
...
-    mode: 0400
-  become: true
-  when:
-    - zabbix_agent_tlspskidentity_file is defined
-    - zabbix_agent_tlspskidentity is defined
-  notify:
-    - restart zabbix-agent
-    - restart win zabbix agent
-    - restart mac zabbix agent
+  when: (zabbix_agent_os_family == "Windows") or (zabbix_agent_docker | bool)
```
Comment (ragingpastry) [inline roles/zabbix_agent/tasks/tlspsk_auto.yml:6]: Correct me if I am wrong, but shouldn't this be an AND condition?

Comment (jeremybusk) [reply roles/zabbix_agent/tasks/tlspsk_auto.yml]: Good find. That should not be there. I did agent2 to but forgot to remove or condition on agentd

File: roles/zabbix_agent/tasks/Windows_conf.yml [hunk 2]
```diff
@@ -13,6 +13,12 @@
     dest: "{{ zabbix_win_install_dir }}\\conf\\{{ zabbix_win_config_name }}"
   notify: restart win zabbix agent
 
+- name: "Windows | Configure zabbix-agent in install_dir root"
```
[the remark this answers is not in the document]
Comment (jeremybusk) [reply roles/zabbix_agent/tasks/Windows_conf.yml:16]: I found unwanted task (it's in Windows_conf.yml) in Windows.yml
```
- name: "Windows | Configure zabbix-agent"
  win_template:
    src:  "{{ zabbix_win_config_name }}.j2"
    dest: "{{ zabbix_win_install_dir }}\\{{ zabbix_win_config_name }}"
  notify: restart win zabbix agent
```
and changed everything to reference {{ zabbix_win_install_dir }}\conf\{{ zabbix_win_config_name }}

**Comment by jeremybusk** [comment]
I think I fixed the issues on this. Ready for another look.

**Comment by dj-wasabi** [comment]
Thanks! 👍

**Pull request merged**